### ISSUE TITLE
PSNR Computation Method

### ISSUE THREAD

**Issue by AshwinKumarVijay** [pytorch/examples#522]
Hello.

I'm having a little difficulty in understanding how you've computed the Average PSNR here. You've taken the average MSE Loss for each batch, but then averaged the PSNR across all the batches. 

Should you compute the Average MSE Loss over all the images, and then the PSNR using that MSE? Or compute the PSNR for each image in the batch separately, and then compute the average PSNR from that?

As an Example:
MSE of Image 1: 0.5
MSE of Image 2: 0.1

Method A (Average MSE):
Average MSE = 0.3
PSNR = 10 * log10(1/0.3) = 5.22879

Method B (Average PSNR):
PSNR = (10 * log10(1/0.5) + 10 * log10(1/0.1)) / 2 = 6.50515

They're not going to provide the same output, so I think the method that you've used is incorrect.

The TensorFlow PSNR function suggests that the way to do it is per image:
"Produces a PSNR value for each image in batch."

https://www.tensorflow.org/api_docs/python/tf/image/psnr

Would that correspond to the following code:

```
   per_image_mse_loss = F.mse_loss(fake, real, reduction='none')
   per_image_psnr = 10 * torch.log10(10 / per_image_mse_loss)
   tensor_average_psnr = torch.mean(per_image_psnr ).item()
```

Could you please let me know if I'm mistaken in some fashion? Because this is a really popular starting point, I wanted to confirm the correct method.

**Comment by Ir1d** [comment]
https://github.com/pytorch/examples/blob/master/super_resolution/main.py#L70

**Comment by fourson** [comment]
The same question about this

**Comment by zakajd** [comment]
> The same question about this

Correct answer is to compute metric first and then average.
You can reference to this[ implementation of PSNR for PyTorch](https://github.com/photosynthesis-team/photosynthesis.metrics) and some other metrics too
